scripts/releaseChangelog.mjs

```diff
@@ -27,147 +25,82 @@ function parseTags(commitMessage) {
-    auth: githubToken,
-    request: {
-      fetch,
-    },
+  const latestTaggedVersion = await findLatestTaggedVersion({
+    cwd: process.cwd(),
+    fetchAll: false,
   });
-
-  const latestTaggedVersion = await findLatestTaggedVersion();
-  const lastRelease = lastReleaseInput !== undefined ? lastReleaseInput : latestTaggedVersion;
-  if (lastRelease !== latestTaggedVersion) {
+  const previousRelease =
+    previousReleaseParam !== undefined ? previousReleaseParam : latestTaggedVersion;
+  if (previousRelease !== latestTaggedVersion) {
     console.warn(
-      `Creating changelog for ${lastRelease}..${release} when latest tagged version is '${latestTaggedVersion}'.`,
+      `Creating changelog for ${previousRelease}..${release} while the latest tagged version is '${latestTaggedVersion}'.`,
     );
   }
 
-  /**
-   * @type {AsyncIterableIterator<Octokit.Response<Octokit.ReposCompareCommitsResponse>>}
-   */
-  const timeline = octokit.paginate.iterator(
-    octokit.repos.compareCommits.endpoint.merge({
-      owner: 'mui',
-      repo,
-      base: lastRelease,
-      head: release,
-    }),
+  const commitsItems = (
+    await fetchCommitsBetweenRefs({
+      lastRelease: previousRelease,
+      release,
+      repo: 'material-ui',
+    })
+  ).filter(
+    (commit) => !commit.author.login.endsWith('[bot]') && !commit.message.startsWith('[website]'),
   );
 
-  /**
-   * @type {Octokit.ReposCompareCommitsResponseCommitsItem[]}
-   */
-  const commitsItems = [];
-  for await (const response of timeline) {
-    const { data: compareCommits } = response;
-    commitsItems.push(...compareCommits.commits.filter(filterCommit));
-  }
-
-  let warnedOnce = false;
-
-  const getAuthor = (commit) => {
-    if (!commit.author) {
-      if (!warnedOnce) {
-        console.warn(
-          `The author of the commit: ${commit.commit.tree.url} cannot be retrieved. Please add the github username manually.`,
-        );
-      }
-      warnedOnce = true;
-      return chalk.red("TODO INSERT AUTHOR'S USERNAME");
-    }
-
-    const authorLogin = commit.author.login;
-
-    if (authorLogin === 'github-actions[bot]') {
-      const authorFromMessage = /\(@(?<author>[a-zA-Z0-9-_]+)\) \(#[\d]+\)/.exec(
-        commit.commit.message.split('\n')[0],
-      );
-      if (authorFromMessage.groups?.author) {
-        return authorFromMessage.groups.author;
-      }
-    }
-
-    return authorLogin;
-  };
-
-  const authors = Array.from(
-    new Set(
-      commitsItems.map((commitsItem) => {
-        return getAuthor(commitsItem);
-      }),
-    ),
-  );
-  const contributorHandles = authors
-    .sort((a, b) => a.localeCompare(b))
-    .map((author) => `@${author}`)
-    .join(', ');
+  const contributorHandles = getAllContributors(commitsItems);
 
   // We don't know when a particular commit was made from the API.
   // Only that the commits are ordered by date ASC
   const commitsItemsByDateDesc = commitsItems.slice().reverse();
```

Comment:
Use `Map` lookup instead of `.indexOf` to turn the complexity from exponential to linear. (`.sort()` is `O(n log(n))` and `.indexOf()` is `O(n)`, `Map.get()` is `O(1)` so I suppose we go from `O(n n log(n))` to `O(n log(n))`)

```suggestion
  const commitsItemsByDateAsc = new Map(commitsItems.map((item, index) => [item, index]));
```

I may be misreading, but if you reverse the operation, then you can also omit the `.reverse()`, right?

```tsx
return commitsItemsByDateDesc.get(b) - commitsItemsByDateDesc.get(a);
```